driver/tests/mpi/test_restart.py

```diff
@@ -0,0 +1,111 @@
+import subprocess
+import unittest.mock
+
+import gt4py
+import numpy as np
+import pytest
+import xarray as xr
+import yaml
+import zarr
+
+import pace.dsl
+from pace.driver import CreatesComm, DriverConfig
+from pace.driver.state import DriverState
+from pace.util.null_comm import NullComm
+
+
+class NullCommConfig(CreatesComm):
+    def __init__(self, layout):
+        self.layout = layout
+
+    def get_comm(self):
+        return NullComm(
+            rank=0,
+            total_ranks=6 * self.layout[0] * self.layout[1],
+            fill_value=0.0,
+        )
+
+    def cleanup(self, comm):
+        pass
+
+
+def test_default_save_restart():
+    driver_config = DriverConfig(
+        stencil_config=pace.dsl.StencilConfig(),
+        nx_tile=12,
+        nz=79,
+        dt_atmos=225,
+        days=0,
+        hours=0,
+        minutes=0,
+        seconds=225,
+        layout=(1, 1),
+        initialization=unittest.mock.MagicMock(),
+        performance_config=unittest.mock.MagicMock(),
+        comm_config=NullCommConfig((1, 1)),
+        diagnostics_config=unittest.mock.MagicMock(),
+        dycore_config=unittest.mock.MagicMock(nwat=6),
+        physics_config=unittest.mock.MagicMock(),
+    )
+    assert driver_config.save_restart is False
+
+
+def test_restart_results():
+    subprocess.call("tests/mpi/run_save_and_load_restart.sh")
```

Comment:
Can you add a `try: finally` block in this test that checks for and deletes the RESTART folder to clean up the test?